src/EA.Weee.RequestHandlers.Tests.DataAccess/AatfReturn/AatfTaskList/FetchObligatedWeeeForReturnDataAccessTests.cs

```diff
@@ -59,46 +62,73 @@ public async Task FetchObligatedWeeeForReturn_ReturnedListShouldContainAllTonnag
+                };
+
+                var obligatedReusedRequest = new AddObligatedReused()
+                {
+                    AatfId = aatf.Id,
                     ReturnId = @return.Id,
                     OrganisationId = organisation.Id,
                     CategoryValues = categoryValues
                 };
 
                 var weeeReceivedAmount = new List<WeeeReceivedAmount>();
+                var weeeReusedAmount = new List<WeeeReusedAmount>();
 
                 foreach (var categoryValue in obligatedReceivedRequest.CategoryValues)
                 {
                     weeeReceivedAmount.Add(new WeeeReceivedAmount(weeeReceived, categoryValue.CategoryId, categoryValue.HouseholdTonnage, categoryValue.NonHouseholdTonnage));
                 }
 
-                var obligatedDataAccess = new AddObligatedReceivedDataAccess(database.WeeeContext);
+                foreach (var categoryValue in obligatedReusedRequest.CategoryValues)
+                {
+                    weeeReusedAmount.Add(new WeeeReusedAmount(weeeReused, categoryValue.CategoryId, categoryValue.HouseholdTonnage, categoryValue.NonHouseholdTonnage));
+                }
 
-                await obligatedDataAccess.Submit(weeeReceivedAmount);
+                var obligateReceivedDataAccess = new AddObligatedReceivedDataAccess(database.WeeeContext);
+                await obligateReceivedDataAccess.Submit(weeeReceivedAmount);
+
+                var obligateReusedDataAccess = new AddObligatedReusedDataAccess(database.WeeeContext);
+                await obligateReusedDataAccess.Submit(weeeReusedAmount);
 
                 var fetchDataAccess = new FetchObligatedWeeeForReturnDataAccess(database.WeeeContext);
 
-                var tonnageList = await fetchDataAccess.FetchObligatedWeeeForReturn(@return.Id);
-                var nonHouseholdList = tonnageList.Select(t => t.NonHouseholdTonnage);
-                var householdList = tonnageList.Select(t => t.HouseholdTonnage);
+                var receivedTonnageList = await fetchDataAccess.FetchObligatedWeeeReceivedForReturn(@return.Id);
+                var receivedNonHouseholdList = receivedTonnageList.Select(t => t.NonHouseholdTonnage);
+                var receivedHouseholdList = receivedTonnageList.Select(t => t.HouseholdTonnage);
+
+                var reusedTonnageList = await fetchDataAccess.FetchObligatedWeeeReceivedForReturn(@return.Id);
+                var reusedNonHouseholdList = reusedTonnageList.Select(t => t.NonHouseholdTonnage);
+                var reusedHouseholdList = reusedTonnageList.Select(t => t.HouseholdTonnage);
 
                 foreach (var category in weeeReceivedAmount)
                 {
-                    nonHouseholdList.Should().Contain(category.NonHouseholdTonnage);
-                    householdList.Should().Contain(category.HouseholdTonnage);
+                    receivedNonHouseholdList.Should().Contain(category.NonHouseholdTonnage);
```

Comment:
So that the test class doesn't get too cumbersome is it possible to split this into two tests, one for the received and one for re-use

Reply:
Split into two classes